server/src/main/java/io/crate/execution/engine/collect/sources/ShardCollectSource.java

```diff
@@ -400,6 +400,10 @@ private CompletableFuture<BatchIterator<Row>> shardFailureFallbackOrRaise(Throwa
             if (Symbols.containsColumn(collectPhase.toCollect(), DocSysColumns.FETCHID)) {
                 throw Exceptions.toRuntimeException(err);
             }
+            // the collect task is not in a resume-able state
+            if (collectTask.completionFuture().isDone()) {
+                throw Exceptions.toRuntimeException(err);
+            }
```

Comment:
I think the fall back function must check if the collect task is resume-able before resuming.

Reply:
Seems like no concrete scenario that will cause this, but could add an assert.

Reply:
Hi @matriv could you have a quick look if it is beneficial to add an assert here? It was my former attempt to https://github.com/crate/crate/pull/15520 before Moll reported steps to reproduce.

Reply:
The assertions are only checked when running tests, and the issue was spotted with running cratedb as "blackbox", so I don't think it will benefit, but on the other hand, the assertion seems logical, so it doesn't harm to add it. Although, the collect task is used only in the else part, shouldn't we have the assertion there?